cards.js

```diff
@@ -92,6 +100,13 @@ class CardSet {
     }
     return cardSet;
   }
+  copy(){
+    let copied=new CardSet();
+    this.forEach(function(card){
+      copied.add(Card.copy(card));
```

Comment:
Nebuutu efektiivaak vienkaarsi izmantot this.toArray un pieskirt to copied.cards?

Reply:
Varētu būt

Reply:
copied vajag uzreiz inicializēt ar to nokopēto kāršu array. Pārrakstīt cardSet cards lauku ir diezgan vardarbīga rīcība.

Reply:
nav atbilstoša konstruktora

Reply:
vispār nav konstruktora